### PR TITLE
Add information to the gdb section of README.md

### PR DESCRIPTION
**Pull request by paulpopelka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
...
+To avoid seeing gdb client messages about this signal it is a good idea to either manually run the "handle SIG63 nostop" command for each debugging
+session or add this command to your ~/.gdbinit file.
+
+If your payload uses fork then the child payload resets all of the gdb related settings so the child payload can't be debugged.  We are working on correcting this shortfall.
+When the child payload finally exec's to a new payload, no gdb related arguments are added to the invocation of km.
+So, there is no debugging in the exec'ed payload either.  We are working on this too.
+
+#### Known gdb related issues
+
+Stack traces are sometimes incomplete.
```
Comment (msterin) [inline README.md:497]: Please use list (start with `* `): 

* one
* two

and maybe just move all known issues (i.e. from the above) here

File: README.md [hunk 2]
```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
...
+gdb non-stop mode (set non-stop on) is not supported.
```
Comment (msterin) [inline README.md:498]: Please emphasize the gdb commands: 
```
* gdb non-stop mode (`set non-stop on`) is not supported. When payload is interrupted, 
ALL threads are stopped for debugging
```

File: README.md [hunk 3]
```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
...
+Do not alter the GS segment selector in your programs.  It is used internally by km.
```
Comment (msterin) [inline README.md:499]: this does not have to be here. At the very least, it should be separate, i.e. 

This does not work in GDB: 
- set non-stop on (btw, what happens if it's typed? ) 
- info registers does not report floating point

Other know issues, FYI :
- GS is used internally, do not alter it in your code 
- dynamically linked payload consumes one FD per lib

**Review by msterin — COMMENTED** [review]
Thanks for the write up. My comments are inline, mainly nits and suggestions

File: README.md [hunk 4]
```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
...
+The simplest flag "-g[port]" tells km to start the gdb server listening on the specified port and then wait for a connection from
+the gdb client.  The default gdb listening port is 2159 (gdbremote from /etc/services).  The km gdb server only uses tcp.
+When you attach to the gdb server the payload is waiting to begin running at the _start entry point in the km payload.
+
+You can also have km's gdb server listening for connections in the background by using the km \-\-gdb-listen command line flag.
+With this flag the payload begins running immediately and the gdb server listens for gdb client connection while the payload runs.
+You can connect to the gdb server, disconnect, and reconnect as often as you want until the payload completes.
+When you connect to the km gdb server all payload threads will be paused until the gdb client starts them using the "cont" or "step" or "next" commands.
+
+You can only connect to a payload's gdb server from the machine hosting the payload.
```
[the remark this answers is not in the document]
Comment (paulpopelka) [reply README.md:483]: You are correct.  We bind to INADDR_ANY so incoming connections from any network interface should work.
The example line from km should not say localhost:2159.  Better to say :2159.
I'll change this document.

File: README.md (hunk 3 above)
Comment (paulpopelka) [reply README.md:499]: Here is what happens if you try to change non-stop:
```
(gdb) show non-stop
Controlling the inferior in non-stop mode is off.
(gdb) set non-stop on
Cannot change this setting while the inferior is running.
(gdb) set non-stop off
Cannot change this setting while the inferior is running.
(gdb)
```

Comment (msterin) [reply README.md:499]: good. I think just adding "Gdb will respond with `Cannot change this setting while the inferior is running.`" to the bullen point above

File: README.md (hunk 2 above)
Comment (paulpopelka) [reply README.md:498]: Removed

File: README.md (hunk 3 above)
Comment (paulpopelka) [reply README.md:499]: Removed

File: README.md [hunk 5]
```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
...
 ```
 
-If you want to run payload and connect gdb to it in-flight, pass `-gdb-listen` to KM.
-
-**TODO**: more documentation and examples here.
+You would then run the following to attach the gdb client to the payload debugger:
+
+```
+[someone@work ~]$ gdb --ex="target remote localhost:2159" ./tests/hello_test.km
+GNU gdb (GDB) Fedora 9.1-5.fc32
```
Comment (msterin) [inline README.md:519]: Please add `-q` flag to gdb , all the licnes/help output from gdb here does not add any value

Comment (paulpopelka) [reply README.md:519]: This is showing what km tells them to do.  km does not tell them to use -q.
Maybe it should be added to what km instructs the user to do.

Comment (msterin) [reply README.md:505]: yes, pls add it in the both places , but for now let's drop it here - it just distracting to have so much to print out

**Review by msterin — APPROVED** [review]

**Pull request merged**